src/complex/Utilities/Parsing/HDF5/H5.hpp

```diff
@@ -1,8 +1,22 @@
 #pragma once
 
-#include <cstdint>
+#include "complex/Common/StringLiteral.hpp"
 
 #define H5_USE_110_API
+#include <H5Opublic.h>
+#include <hdf5.h>
+
+#ifdef H5Support_USE_MUTEX
+#include <mutex>
+#define H5SUPPORT_MUTEX_LOCK()                                                                                                                                                                         \
+  std::mutex mutex;                                                                                                                                                                                    \
+  std::lock_guard<std::mutex> lock(mutex);
+#else
+#define H5SUPPORT_MUTEX_LOCK()
+#endif
```

Comment:
Do we want HDF5 headers included here? I believe this file is analogous to the one in H5Support that we made to avoid recompiling everything when we made small HDF5 changes.

Reply:
Yes. We need the HDF5 headers in here because we are setting an HDF5 Macro to a specific value. If you get the headers out of order compilation fails in with very strange errors. By forcing developers to just #include this file we ensure that we get it correct.

Thinking more through the problem we might want to actually have the HDF5 macro that defines that we are using the 1.10 API as an actual compilation argument -D than sitting in a header file. This will guarantee the problem expansion of macros during compilation.

Reply:
It should be pointed out that this macro does not do what I believe you think it does. If this is added to a function, each call gets its own mutex, making it impossible for two calls to the same function to lock the same mutex.

No actual HDF5 code is supposed to be in this file. Otherwise, you might as well delete H5.hpp and merge it all into H5Support.hpp. If you need HDF5 includes, include H5Support. If all you need are aliases, include H5.hpp.

And if you want this macro to be capable of locking across calls, change it to `static std::mutex mutex;` and it will lock when the same function is called. Move `std::mutex mutex` out of `H5SUPPORT_MUTEX_LOCK` and rename it to something unique if you want it to lock across any function call that contains this macro.